src/test/unit_tests/test_s3_ops.js

```diff
@@ -11,6 +11,7 @@ const http_utils = require('../../util/http_utils');
 const mocha = require('mocha');
 const assert = require('assert');
 const querystring = require('querystring');
+const P = require('../../util/promise');
```

Comment:
as we are trying not to use bluebird, I think we can change that to:
```suggestion
const util = require('util');
```
@guymguym what do you think?

Reply:
You're right @liranmauda, but I want to focus this review. Resolving and we'll handle P.delay across the code later